build_modules/lib/src/meta_module.dart

```diff
@@ -238,14 +238,13 @@ MetaModule _coarseModulesForLibraries(
   var librariesByDirectory = <String, Map<AssetId, ModuleLibrary>>{};
   for (var library in libraries) {
     final dir = _topLevelDir(library.id.path);
-    if (!librariesByDirectory.containsKey(dir)) {
-      librariesByDirectory[dir] = <AssetId, ModuleLibrary>{};
-    }
-    librariesByDirectory[dir]![library.id] = library;
+    (librariesByDirectory[dir] ??= {})[library.id] = library;
```

Comment:
we could use `putIfAbsent` also, but 🤷‍♂️

Reply:
Yeah, I had thought about that pattern too but for some reason `??=` looked nicer to me.

Maybe it was something like dart2js emitting better code for `??=` than `putIfAbsent`? (even though this will not be compiled with dart2js)

Reply:
I would usually use `putIfAbsent`, it's true that this doesn't require a closure though. I am fine with either way.